client/src/sections/courses/LearningSection/index.tsx

```diff
@@ -11,40 +10,33 @@ import FilterIcon from '@/src/shared/icons/FilterIcon';
 import ProgressPercentage from '@/src/shared/components/ProgressPercentage';
 import { useGetLearnerQuery } from '@/services/traineeAPI';
 import { useRouter } from 'next/router';
+import { useAppDispatch } from '@/app/hooks';
+import { type Trainee, addTrainees, seeMoreTrainees } from '@/features/course/learnerSlice';
+import { useSelector } from 'react-redux';
+import { type RootState } from '@/app/store';
 
 const LearningSection: React.FC = () => {
-  const [dataLimiter, setDataLimiter] = useState(10);
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const [selectedSortOption, setSelectedSortOption] = useState('');
 
+  const dispatch = useAppDispatch();
   const router = useRouter();
   const params = router.query;
   const courseID = params.id;
+  const { trainees, isEnrollee, page } = useSelector((state: RootState) => state.courseLearner);
+  const { data: trainee, isLoading } = useGetLearnerQuery({
+    courseId: courseID,
+    isEnrolled: isEnrollee,
```

Comment:
```suggestion
    isEnrolled: true,
```
You can actually give this a static value, because its always true on this page, and also I see its causing some bugs in the page, like flickers and also changing visibility of `see more learners` button when modal is open

Please do the same for modal as well, I think you do not need to store that `isEnrolled` var in the slice